src/index.rs

```diff
@@ -396,7 +389,40 @@ impl Index {
   }
 
   pub(crate) fn update(&self) -> Result {
-    Updater::update(self)
+    let mut updater = Updater::new(self)?;
+
+    loop {
+      match updater.update_index() {
+        Ok(ok) => return Ok(ok),
+        Err(err) => {
+          if let Some(&UpdaterError::Reorged(height)) = err.downcast_ref::<UpdaterError>() {
+            log::info!("{}", err.to_string());
+            self.handle_reorg(height)?;
+            updater = Updater::new(self)?;
+          } else {
+            return Err(err);
+          }
+        }
+      }
+    }
+  }
+
+  pub(crate) fn handle_reorg(&self, height: u64) -> Result {
+    log::info!("rolling back database after reorg at height {}", height);
+    let mut wtx = self.begin_write()?;
+
+    let oldest_savepoint =
```

Comment:
For this, an alternative to consider is to rollback to the latest savepoint and delete it. That would only roll you back 1-4 blocks instead of 10-14. If it's not far enough, then when the updater runs again it will hit a reorg again and rollback to the older savepoint automatically.